app/services/action_handlers/pull_request.rb

```diff
@@ -34,8 +34,12 @@ def review_request_removed
     def review_requested
       pr_data = @payload['pull_request']
       owner = find_or_create_user(pr_data['user'])
-      reviewer = find_or_create_user(pr_data['requested_reviewers'].first)
-      @entity.review_requests.create!(owner: owner, reviewer: reviewer)
+      pr_data['requested_reviewers'].each do |raw_reviewer|
+        reviewer = find_or_create_user(raw_reviewer)
+        @entity.review_requests.create!(owner: owner, reviewer: reviewer)
+      end
+    rescue ActiveRecord::RecordInvalid => e
+      raise unless e.message == 'Validation failed: Pull request has already been taken'
```

Comment:
Some questions:
1. Should the rescue be inside of the `pr_data['requested_reviewers'].each` block? So that it continues with the rest of the elements?
2. If it already exists, don't you want to update it? If yes, you can use Rails 6 new method `create_or_find_by` https://blog.bigbinary.com/2019/03/25/rails-6-adds-create_or_find_by.html

Reply:
`create_or_find_by` won't cost more memory by finding the record? And for the first question, yes it could go inside the block.

Reply:
maybe.. if you don't need to update the object if already created (do you?) then you don't need to use `create_or_find_by`

Reply:
Do you what? I will update it, 'cause I realize that the reviewer could be removed and the state would be 'removed', so I will update the state to 'active'.